lib/fsm/utils.go

```diff
@@ -216,6 +218,25 @@ func ClusterKey(plan storage.OperationPlan) ops.SiteKey {
 	}
 }
 
+// CompleteOperation completes the operation given by the plan in the specified operator.
+// planErr optionally specifies the error to record in the failed message
+func CompleteOperation(plan *storage.OperationPlan, operator ops.Operator, planErr string) (err error) {
```

Comment:
I'd consider renaming this to CompleteOrFailOperation. When I was looking at where it's called above I initially assumed it always sets operation to "completed".